### PR TITLE
Added docker file, brought in routing and added pages

### PR DESCRIPTION
**Pull request by nabeelmsft**
Added docker file, brought in routing and added pages' structure.

### CODE REVIEW

File: agora/main_ui/src/Dockerfile
Comment (fcabrera23) [inline agora/main_ui/src/Dockerfile:1]: Please use the correct golden images. Sharing a MSFT supported version


﻿# build environment
FROM mcr.microsoft.com/cbl-mariner/base/nodejs:18 as build
WORKDIR /app
COPY . .
RUN npm install --production
RUN npm run build

# production environment
FROM mcr.microsoft.com/cbl-mariner/base/nginx:1

RUN tdnf install awk -y 

RUN mkdir -p /usr/share/nginx/html
COPY .env /usr/share/nginx/html
COPY ./nginx.conf.template /etc/nginx/nginx.conf.default
COPY --from=build /app/build /usr/share/nginx/html
COPY ./env-config.sh /docker-entrypoint.d/env-config.sh
RUN chmod +x /docker-entrypoint.d/env-config.sh

EXPOSE 80
#CMD ["nginx", "-g", "daemon off;"]
#CMD ["tail", "-f", "/dev/null"]
CMD ["/bin/sh", "-c", "/docker-entrypoint.d/env-config.sh && nginx -g 'daemon off;'"]

**Review by fcabrera23 — CHANGES_REQUESTED** [review]
@nabeelmsft  please review the new dockerfile

File: agora/main_ui/src/Dockerfile
Comment (nabeelmsft) [reply agora/main_ui/src/Dockerfile:1]: Thanks @fcabrera23 for the feedback. I am bringing in your suggested changes into the Dockerfile. Thank  you for that. Quick question: For the npm, do I need to change anything in your dockerfile? More specifically, I need to run npm start command. Can I add the CMD ["npm", "start"] at the end of your docker file?

Comment (nabeelmsft) [reply agora/main_ui/src/Dockerfile:1]: I have added the docker file with the new check in  https://github.com/nabeelmsft/jumpstart-apps/commit/75679c27c951dc635db219df177c0195d589166c . Thank you for that. Please let me know if I need to make any changes for running the npm start as part of the command.

**Pull request merged**